web/src/main/java/org/springframework/security/web/header/writers/ClearSiteDataHeaderWriter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.header.writers;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.security.web.header.HeaderWriter;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.util.Assert;
+
+/**
+ * Provides support for <a href="https://w3c.github.io/webappsec-clear-site-data/">Clear
+ * Site Data</a>.
+ *
+ * <p>
+ * Developers may instruct a user agent to clear various types of relevant data by delivering
+ * a Clear-Site-Data HTTP response header in response to a request.
+ * <p>
+ *
+ * <p>
+ * Due to <a href="https://w3c.github.io/webappsec-clear-site-data/#incomplete">Incomplete Clearing</a>
+ * section the header is only applied if the request is secure.
+ * </p>
+ *
+ * @author Rafiullah Hamedy
+ * @since 5.2.0
+ */
+public class ClearSiteDataHeaderWriter implements HeaderWriter {
```

Comment:
Let's please make this class `final` for now, favoring composition over inheritance. We can open it up later if we need to.